### PR TITLE
cpu/stm32: removed file guards from periph drivers

### LINKED ISSUE #7981: periph: remove file guards from peripheral drivers
follow up on #7241

Since #7241 was merged, there is no need anymore for ugly file guards around peripheral driver implementations, as they are now finally only touched by the compiler if they are actually (a) included in the build (using USEMODULE) and (b) provided by a board/cpu (as in FEATURES_PROVIDED). So now it is time to clean up a little bit, and this issue should help so we don't forget anything.

Removed file guards from peripheral drivers in:
- [x] arm7_common
- [x] atmega1281
- [x] atmega2560
- [x] atmega328p
- [x] atmega_common (#8959)
- [x] cc2538 (#8955)
- [x] cc26x0
- [x] cc430
- [x] cortexm_common
- [x] efm32
- [x] ezr32wg
- [x] k22f
- [x] k60
- [x] k64f
- [x] kinetis_common (-> @gebart, #7882)
- [x] kw2xd
- [x] lm4f120 (#8958)
- [x] lpc1768 (#8956) (-> @aabadie, #8956)
- [x] lpc2387 (#8957)
- [x] mips32r2_common
- [x] mips_pic32_common
- [x] mips_pic32mx
- [x] mips_pic32mz
- [x] msp430_common
- [x] msp430fxyz
- [x] native
- [x] nrf51 (-> @haukepetersen, #7982)
- [x] nrf52 (-> @haukepetersen, #7982)
- [x] nrf5x_common (-> @haukepetersen, #7982)
- [x] sam0_common ( -> @haukepetersen, #7984)
- [x] sam3 ( -> @haukepetersen, #7984)
- [x] sam_common
- [x] samd21 ( -> @haukepetersen, #7984)
- [x] saml21 ( -> @haukepetersen, #7984)
- [x] stellaris_common
 - [x] stm32_common (-> @haukepetersen, #7983)
 - [x] stm32f0 (-> @haukepetersen, #7983)
 - [x] stm32f1 (-> @haukepetersen, #7983)
 - [x] stm32f2 (-> @haukepetersen, #7983)
 - [x] stm32f3 (-> @haukepetersen, #7983)
 - [x] stm32f4 (-> @haukepetersen, #7983)
 - [x] stm32f7 (-> @haukepetersen, #7983)
 - [x] stm32l0 (-> @haukepetersen, #7983)
 - [x] stm32l1 (-> @haukepetersen, #7983)
 - [x] stm32l4 (-> @haukepetersen, #7983)
- [x] x86

### PR DESCRIPTION
**Pull request by haukepetersen**
addresses #7981
same as #7982

### CODE REVIEW

File: cpu/stm32_common/periph/dac.c [hunk 1]
```diff
@@ -25,9 +25,6 @@
 #include "assert.h"
 #include "periph/dac.h"
 
-/* only compile this, if the CPU has a DAC */
```
[the remark this answers is not in the document]
Comment (haukepetersen) [reply cpu/stm32_common/periph/dac.c]: no, why? The stm does not have this features, so it should not be build for this cpu in the first place..

File: cpu/stm32_common/periph/pwm.c [hunk 2]
```diff
@@ -29,8 +29,6 @@
 #include "periph/pwm.h"
 #include "periph/gpio.h"
 
-#ifdef PWM_NUMOF
```
[the remark this answers is not in the document]
Comment (haukepetersen) [reply cpu/stm32_common/periph/pwm.c]: same here.

**Comment by haukepetersen** [comment]
Murdock does not seem to have trouble building for the `iotlab-m3` nor the `nucleo-f103`. Could it be that this is a local problem on your machine?

**Comment by aabadie** [comment]
> Murdock does not seem to have trouble building for the iotlab-m3 nor the nucleo-f103. Could it be that this is a local problem on your machine?

Ah yes, sorry, I totally misunderstood that. It's fine.

**Comment by aabadie** [comment]
Just need rebase now that #7864 is merged

**Comment by haukepetersen** [comment]
rebased and fixed two cppcheck issues with i2c drivers that murdock pointed out (unrelated to the actual PR, but needed to make Murdock happy :-)).

**Comment by aabadie** [comment]
Before merging, I'd like to give a try of the I2C drivers modified.

**Pull request merged**